### ISSUE TITLE
Steam Performance is Very Slow

### ISSUE THREAD

**Issue by genX75** [ValveSoftware/steam-for-linux#4653]
Here is a video outlining the performance issues: https://youtu.be/UKT3OI-0szQ

I have tried the mainline and beta versions of steam. Both versions produce the same performance problems.

Perhaps there are all ready open issues about this but I wanted to give it more attention.

My system information: http://pastebin.ubuntu.com/23300385/

**Comment by genX75** [comment]
I opened a bug report with chromium here: https://bugs.chromium.org/p/chromium/issues/detail?id=654423
which is merged into: https://bugs.chromium.org/p/chromium/issues/detail?id=638667

I think this is the crux of the problem. Something is causing X to slow to a crawl. Which sounds driver related? Anyhow thanks guys!

**Comment by genX75** [comment]
I was able to fix the issue. I need more time to pinpoint the exact causes. I'll report my findings soon.

**Comment by ekaris** [comment]
Are you using a wayland session? That would cause a slowdown like you're describing.

**Comment by blu132** [comment]
I am having this same issue except it's even worse.  As long as steam is even launched, other applications slow down considerably, even when the steam client window is minimized. The redraw on all open windows is so slow, you can see X writing the text into the window decoration.  Libre Office Writer takes many seconds to even draw its own interface and even low ressource applications like xfig or xpdf or the CDE File Manager dtfile slow down to speeds unknown before. The computer in question has an AMD Phenom II Quad core and two 5770 Ati cards in cross fire. While this is not that powerful, it should be enough by far for a client program, that claims to work on a 1 Ghz Machine with 512 MB of RAM...

Operating system is Devuan GNU/Linux 1.0 jessie, which is mostly equivalent to Debian 8, except it doesn't use systemd.  The steam installed is the standard .deb without modifications, patched to the current non-beta version.

This issue is worse with CDE and less terrible while running i3wm, but in both cases, makes some programs with more complex interfaces unusable.

**Comment by genX75** [comment]
I was not able to actually fix the issue. It is definitely chromium making far too many draw calls. I talked to the chromium developers and they told me linux is not a priority because no one uses it. When steam was first created for linux the performance was fantastic. When they made the switch to aura the performance on linux was neglected. Perhaps firefox is a better option? Rather sad really. In my spare time I'm reading the source code to maybe implement a patch for better scheduling.

**Comment by knerlington** [comment]
The Steam client itself causes this issue. Just by having it running every other window from other applications become extremely slow when it comes to window resizing. I can't resize the main Steam window, the friends list or even nautilus windows without having a extreme delay. Note that this happens even without any other browser running at all. While Chromium and Firefox have done the same for me in the past they don't as of now. If Steam isn't running resizing of random windows is smooth and responsive as it should be, but as soon as I run Steam.... Well you get the point.

Ubuntu 16.04.4 LTS with stock kernel 4.13.0-38-generic
Fully up to date.
GTX 1070 with proprietary driver 390.48 or 384.xx
6700K @ Stock
16GB DDR4

**Comment by kisak-valve** [comment]
Hello @genX75, @knerlington, are either of you still experiencing this issue on an up to date system?

**Comment by genX75** [comment]
@kisak-valve Yes steam still struggles to repaint while resizing with a ryzen cpu @ 4ghz. I moved away from chrome completely because of it. I would love to be able to test it with an AMD card like a vega 64 one day to see if its isolated to nvidia or not.
